baremetal_network_provisioning/drivers/hp/hp_snmp_provisioning_driver.py

```diff
@@ -56,11 +58,48 @@ class HPSNMPProvisioningDriver(api.NetworkProvisioningApi):
 
     def __init__(self):
         """initialize the snmp driver."""
+        self.conf = cfg.CONF
+        self.bnp_sync_enable = self.conf.default.bnp_sync_enable
+        self.bnp_sync_interval = float(
+            self.conf.default.get('bnp_sync_interval'))
+        if self.bnp_sync_enable:
+            self.start_snmp_polling()
         # TODO(selva) need to check how we can load dynamically
         drvr = 'baremetal_network_provisioning.drivers.snmp_driver.SNMPDriver'
         self.context = neutron_context.get_admin_context()
         self._load_drivers(drvr)
 
+    def _snmp_sync_thread(self):
+        """Sync switch database periodically."""
+        LOG.info(_LI('BMNP start thread....'))
+        while True:
+            portmaps = db.get_all_bnp_swport_mappings(self.context)
+            for portmap in portmaps:
+                swport = db.get_bnp_phys_switch_port_by_id(
+                    self.context, portmap['switch_port_id'])
+                switch = db.get_bnp_phys_switch(self.context,
+                                                portmap['switch_id'])
+                try:
+                    snmp_drv = discovery_driver.SNMPDiscoveryDriver(switch)
+                    port_status = snmp_drv.get_port_status(swport['ifindex'])
+                except Exception as e:
+                    LOG.error(_LE("Exception: %s"), e)
+                    if swport['port_status'] != 'UNKNOWN':
+                        db.update_bnp_phys_swport_status(
+                            self.context, swport['switch_id'],
+                            swport['interface_name'], 'UNKNOWN')
+                else:
+                    status = constants.PORT_STATUS.get(str(port_status))
+                    if swport['port_status'] != status:
+                        db.update_bnp_phys_swport_status(
+                            self.context, swport['switch_id'],
+                            swport['interface_name'], status)
+            eventlet.sleep(self.bnp_sync_interval)
+
+    def start_snmp_polling(self):
+        """Spawn a thread to poll the switch db."""
+        eventlet.spawn(self._snmp_sync_thread)
+
     def create_port(self, port):
```

Comment:
Already up-to-date. There are two functions get_bnp_phys_port_by_id and get_bnp_phys_switch_port_by_id which are doing same. So I deleted get_bnp_phys_port_by_id and used get_bnp_phys_switch_port_by_id  every where.